src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/ChecklistServiceImpl.java

```diff
@@ -59,10 +66,202 @@ public ChecklistServiceImpl(ApplicationData applicationData, Tools tools, InforW
+														getResult.getResultData().getWorkOrderActivityCheckListDefault());
+
+		workOrderActivityCheckListDefaultResult.setUserResponsibilities(
+				getResult.getResultData().getWorkOrderActivityCheckListDefault().getUSERRESPONSIBILITY());
+		WorkOrderActivityChecklistSignatureResult[] res = filterSignatures(workOrderActivityCheckListDefaultResult);
+		getResponsibilityDescriptions(context, res);
+		return res;
+	}
+
+	private void getResponsibilityDescriptions(InforContext context, WorkOrderActivityChecklistSignatureResult[] signatures)
+			throws InforException {
+
+		GridRequest gridRequest = new GridRequest("LVUSERRESPONSIBILITIES");
+		gridRequest.setDataspyID("4297");
+		gridRequest.getParams().put("param.rentity", "RESP");
+		List<GridRequestFilter> filters = new LinkedList<GridRequestFilter>();
+		Map<String, List<WorkOrderActivityChecklistSignatureResult>> responsibilityToSignature
+				= new HashMap<>();
+		for(WorkOrderActivityChecklistSignatureResult signatureResult : signatures){
+			String responsibilityCode = signatureResult.getResponsibilityCode();
+			if(responsibilityCode != null) {
+				List<WorkOrderActivityChecklistSignatureResult> signaturesWithResp =
+						responsibilityToSignature.getOrDefault(responsibilityCode, new ArrayList<WorkOrderActivityChecklistSignatureResult>());
+				signaturesWithResp.add(signatureResult);
+				responsibilityToSignature.put(responsibilityCode, signaturesWithResp);
+				filters.add(new GridRequestFilter(
+						"responsibility", responsibilityCode, "=", GridRequestFilter.JOINER.OR, false, false));
+			}
+		}
+		if(filters.size() == 0) return;
+		gridRequest.setGridRequestFilters(filters);
+		GridRequestResult request = gridsService.executeQuery(context, gridRequest);
+		GridRequestRow[] rows = request.getRows();
+		GridRequestCell emptyCell = new GridRequestCell("", "");
+		BiFunction<GridRequestRow, String, String> getCellContent = (row, tag)->
+					Arrays.stream(row.getCell())
+							.filter(cell -> cell.getTag().equals(tag)).findFirst().orElse(emptyCell).getContent();
+
+		Arrays.stream(rows).forEach(row ->
+				responsibilityToSignature.get(getCellContent.apply(row, "responsibility")).stream()
+						.forEach(signature -> signature.setResponsibilityDescription(getCellContent.apply(row, "description"))));
+	}
+
+	private WorkOrderActivityChecklistSignatureResult[] filterSignatures(WorkOrderActivityCheckListDefaultResult workOrderActivityCheckList) throws Exception {
+		DateAdapter dateAdapter = new DateAdapter();
+		String reviewerQualification = workOrderActivityCheckList.getReviewerQualification();
+		String performer1Qualification = workOrderActivityCheckList.getPerformer1Qualification();
+		String performer2Qualification = workOrderActivityCheckList.getPerformer2Qualification();
+		List<String> qualifications = new ArrayList<>();
+		for(USERDEFINEDCODEID_Type userData : workOrderActivityCheckList.getUserResponsibilities())
+			qualifications.add(userData.getUSERDEFINEDCODE());
```

Comment:
Is WorkOrderActivityCheckListDefaultResult::getUserResponsibilities returning a List<USERDEFINEDCODEID_Type> optimal? What if we wish to use this method elsewhere, will we have to convert it to a List<String> every time? Should this not be done when creating the instance of the class instead? Using an InforField annotation might not necessarily be appropriate here.